src/posit/connect/resources.py

```diff
@@ -203,26 +204,17 @@ def find(self, uid) -> T:
         Returns
         -------
         T
-
-        Raises
-        ------
-        ValueError
-            If no record is found.
         """
-        # todo - add some more comments about this
-        if self._cache:
+        if self.cached():
             conditions = {self._uid: uid}
             result = self.find_by(**conditions)
-        else:
-            endpoint = self._endpoint + uid
-            response = self._ctx.session.get(endpoint)
-            result = response.json()
-            result = self._create_instance(**result)
-
-        if not result:
-            raise ValueError(f"Failed to find instance where {self._uid} is '{uid}'")
+            if result:
+                return result
 
-        return result
+        endpoint = self._ctx.url + self._path + uid
+        response = self._ctx.session.get(endpoint)
+        result = response.json()
+        return self._to_instance(result)
```

Comment:
Note that I removed the call to invalidate the cache that existed a few commits before. After some additional consideration, I concluded that invalidating the cache is an unwanted side effect.

I think there is still an argument for invalidating the cache or appending the instance to the cached list. But, I don't think we have a good enough understanding of the side effects to proceed with either implementation.

Reply:
Why not just call `.find_by()` all the time? 

If the cache exists, `._data` returns quickly. If not, it asks the server.

Then both methods have the same quirks. (Where as `find()` will not alter the cache, but find_by will... causing a followup call to `find()` to use the cached values, behaving differently.

Reply:
I think in either situation, we end up with conflicting ideas. 

If we always depend on `find_by`, we must fetch the entire collection before returning a single value, which would take significantly longer than a single GET request for the value.

Today, I think the obvious solution would be to **always** call the HTTP GET method to get the value from the server. This will sometimes be slightly slower than an in-memory list scan. But in reality, it's going to be a negligible difference. The weird edge case with this solution is when another process creates the value fetched by GET *after* the `_cache` is set. In this situation, the value returned by `find` will exist on the server but not in the _cache. This would probably warrant a cache invalidation. But that would take extra time to compute and may not be consistent behavior across all endpoints.

tl;dr - the speed up via `find_by` probably isn't worth the trouble. Classic over engineering.